### PR TITLE
Fix shift operators byte value space inconsistencies

### PR DESCRIPTION
**Pull request by Kishanthan**
This PR updates bitwise shift operator logic to properly handle byte value space. Introduced few instruction codes to differentiate byte related shift operations with int.

### CODE REVIEW

**Comment by lankavitharana** [comment]
```ballerina

import ballerina/io;

function main(string... args) {
    byte a = 129;
    byte c = a << 1;
    byte d = c >> 1;
    io:println(a);
    io:println(c);
    io:println(d);
}
```

This will print 
129
2
129

IMO last println should print "1" as a result

File: tests/ballerina-unit-test/src/test/resources/test-src/types/byte/byte-value.bal [hunk 1]
```diff
@@ -230,23 +230,44 @@ function testBitwiseXorOperator(byte a, byte b, int i, int j) returns (byte, int
...
+    byte r3 = a >>> j;
     return (r1, r2, r3);
 }
+
+function testBitwiseUnsignedRightShiftOperator2(byte a, byte b, int i, int j) returns (int, int, int){
+    int r1 = <int> (a >>> b);
+    int r2 = i >>> j;
+    int r3 = <int> (a >>> j);
+    return (r1, r2, r3);
+}
```
Comment (lankavitharana) [inline tests/ballerina-unit-test/src/test/resources/test-src/types/byte/byte-value.bal]: please add newline at end

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/codegen/InstructionCodes.java [hunk 2]
```diff
@@ -248,6 +249,8 @@
     int ARRAYLEN = 207;
     int LENGTHOF = 208;
 
+    int BIUNRSHIFT = 209;
```
Comment (lankavitharana) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/util/codegen/InstructionCodes.java]: shall we remove these?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 3]
```diff
@@ -1307,15 +1307,9 @@ public void visit(BLangBinaryExpr binaryExpr) {
...
-                binaryExpr.rhsExpr = createTypeConversionExpr(binaryExpr.rhsExpr, binaryExpr.rhsExpr.type,
-                        symTable.intType);
-            }
-            if (TypeTags.BYTE == binaryExpr.lhsExpr.type.tag) {
-                binaryExpr.lhsExpr = createTypeConversionExpr(binaryExpr.lhsExpr, binaryExpr.lhsExpr.type,
-                        symTable.intType);
-            }
+        if (isBitwiseShiftOperation(binaryExpr) && TypeTags.BYTE == binaryExpr.rhsExpr.type.tag) {
+            binaryExpr.rhsExpr = createTypeConversionExpr(binaryExpr.rhsExpr, binaryExpr.rhsExpr.type,
+                    symTable.intType);
```
Comment (lankavitharana) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: shall we return from here?

Comment (Kishanthan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: done

**Pull request merged**